packages/sign-client/src/controllers/engine.ts

```diff
@@ -674,6 +675,7 @@ export class Engine extends IEngine {
     const { id, params } = payload;
     try {
       this.isValidRequest({ topic, ...params });
+      await this.client.pendingRequest.set(payload.id, { id, topic, params });
```

Comment:
Lifecycle: Is there a risk of a bunch of pendingRequests being set but never cleaned up if they're never responded to? Could be used to create bloat either accidentally or maliciously.

Reply:
Might be worth adding some kind of clean-up fx on start where expired requests are removed if not responded to just to be sure

Reply:
Hmm can we handle the same way we handle proposals, i.e. by setting it in the expirer which will then take care of it (once we make sure it's handled on `EXPIRER_EVENTS.expired`):
https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/packages/sign-client/src/controllers/engine.ts#L344

It's the same scenario but it's a pending request instead of a pending session settlement.

Reply:
added to the expirer flow. 
Question: If we are to add more stuff to the expirer might be worth adding new field `type` so the filtering in each listener  would be more clear/easier. Wdyt?

Reply:
Nice thank you 💯 

> Question: If we are to add more stuff to the expirer might be worth adding new field type so the filtering in each listener would be more clear/easier. Wdyt?

Yeah definitely, it was already overly implicit (deciding if its a session or proposal based on whether it has an id or topic). Great candidate for some refactoring 👍